core/stdlib/std.ncl

```diff
@@ -2670,13 +2670,19 @@
       "%
       = fun regex => %str_is_match% regex,
 
+    Match
+      | doc m%"
+        Result of regular expression search on a string.
+        `matched` is the part of the string that matches,
+        `index` is the index of the first character that was part of the match in the string,
+        `groups` is an array of all capture groups if there were any.
+      "%
+      = { matched | String, index | Number, groups | Array String },
+
     find
-      : String -> String -> { matched : String, index : Number, groups : Array String }
+      | String -> String -> Match
```

Comment:
I wanted to de-duplicate the match record type. Shall I keep `Match` or simply copy the type annotation?

Reply:
Also, I had to use a contract application instead of a type annotation because otherwise the typechecking test fails.

Reply:
But it's used only one place, if I'm not mistaken, as you don't use it in `find_all` anyway. I would vote for copying the type annotation, to keep the static type annotation here (in particular this optimizes the return contract of `find` away since #1671)

Reply:
Oh my bad. I meant to use it in `find_all` too but I must've forgotten 😅. I'll remove it then.